builtin_string.go

```diff
@@ -952,58 +952,61 @@ func (r *Runtime) createStringIterProto(val *Object) objectImpl {
 	return o
 }
 
+func (r *Runtime) createString(val *Object) objectImpl {
```

Comment:
hmm yeah, i think ordinarily, it's called with the object's current `.val` property like [this](https://github.com/andrewzlchen/goja/blob/59b820598f8158c8c4029e3502ec4ad2d4f4628f/object_lazy.go#L15)

I realized yesterday that we were instantly instantiating the `String` namespace because of [this line ](https://github.com/mongodb-forks/goja/blob/59b820598f8158c8c4029e3502ec4ad2d4f4628f/builtin_string.go#L1009)which eventually makes the lazy object `r.globals.String` instantiate itself

I think this is the reason for the unexpected performance degradation because were weren't getting any benefit from lazy loading `String`